tenacity/_asyncio.py

```diff
@@ -60,6 +61,47 @@ async def __call__(  # type: ignore[override]
             else:
                 return do  # type: ignore[no-any-return]
 
+    @classmethod
+    def _wrap_action_func(cls, fn: t.Callable[..., t.Any]) -> t.Callable[..., t.Any]:
+        if _utils.is_coroutine_callable(fn):
+            return fn
+
+        async def inner(*args: t.Any, **kwargs: t.Any) -> t.Any:
+            return fn(*args, **kwargs)
+
+        return inner
+
+    def _add_action_func(self, fn: t.Callable[..., t.Any]) -> None:
+        self.iter_state.actions.append(self._wrap_action_func(fn))
```

Comment:
Looks like this is the same as the one from `BaseRetrying`, no?

Reply:
This one is slightly different, as it wraps the `fn` function with `self._wrap_action_func()`, which makes sure all calls are async.

I considered adding a separate `_wrap_action_func()` in `BaseRetrying` which could just `return fn` and override it for `asyncio`, to keep the `append()` logic just in `BaseRetrying`, but seemed a bit of an overkill considering it's just a one liner. Would love to hear your thoughts on it though.

Reply:
Sounds good enough.